text/0436-gamelift-l2.md

```diff
@@ -0,0 +1,983 @@
+such as `metricCurrentTickets`, or `metricMatchAccepted` (see [`IMatchmakingConfiguration`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-gamelift.IMatchmakingConfiguration.html)
+for a full list). CDK also provides a generic `metric` method that can be used to produce metric configurations for any metric provided by GameLift FlexMatch; the configurations are pre-populated with the correct dimensions for the matchmaking configuration.
+
+```ts fixture=with-matchmaking-configuration
+import * as cloudwatch from '@aws-cdk-lib/aws-cloudwatch';
+// Alarm that triggers when the per-second average of not placed matches exceed 10%
+const matchesPlacedRatio = new cloudwatch.MathExpression({
+  expression: '1 - (matchesPlaced / matchedCreated)',
+  usingMetrics: {
+    matchesPlaced: matchmakingConfiguration.metricMatchesPlaced({ statistic: cloudwatch.Statistic.SUM }),
+    matchesCreated: matchmakingConfiguration.metric('MatchesCreated'),
+  },
+});
+new Alarm(this, 'Alarm', {
+  metric: matchesPlacedRatio,
+  threshold: 0.1,
+  evaluationPeriods: 3,
+});
+```
+
+See: [Monitoring Using CloudWatch Metrics](https://docs.aws.amazon.com/gamelift/latest/developerguide/monitoring-cloudwatch.html)
+in the *Amazon GameLift Developer Guide*.
+
+#### GameLift Hosting
+
+##### Defining a GameLift Fleet
+
+GameLift helps you deploy, operate, and scale dedicated game servers for session-based multiplayer games. It helps you regulate the resources needed to host your games, finds available game servers to host new game sessions, and puts players into games.
+
+###### Creating a realtime game server fleet
+
+This lightweight server solution provides ready-to-go game servers that you can configure to fit your game. To set up and optionnally customize a realtime server fleet, you need to provide a script (in the form of some JavaScript code).
+
+```ts
+import * as s3 from 'aws-cdk-lib/aws-s3-assets';
+import * as gamelift from 'aws-cdk-lib/aws-gamelift';
+
+// Script can be declared using either declarative version in the constructor
+const script = new gamelift.Script(this, 'Realtime script', {
+  location: new s3.Asset(this, "SampleScriptAsset", {
+    path: path.join(__dirname, 'file-asset.js')
+  })
+});
+
+// Either using dedicated factory static method
+const script = Script.fromAsset(path.join(__dirname, 'file-asset.js');
+
+new gamelift.Fleet(this, 'Realtime server fleet', {
+  script: script
+});
+```
+
+###### Creating a custom game server fleet
+
+Your uploaded game servers are hosted on GameLift virtual computing resources, called instances. You set up your hosting resources by creating a fleet of instances and deploying them to run your game servers. You can design a fleet to fit your game's needs.
+
+```ts
+import * as s3 from 'aws-cdk-lib/aws-s3-assets';
+import * as gamelift from 'aws-cdk-lib/aws-gamelift';
+
+const build = new gamelift.Build(this, 'Game server build', {
```

Comment:
The `fromAsset()` pattern can be used here as well.

Reply:
perfect